R/hiper_glm.R

```diff
@@ -3,25 +3,34 @@ hiper_glm <-
   function(design,
            outcome,
            model = "linear",
-           option = list(mle_solver = c("pseudo inverse", "BFGS"))) {
-    supported_model <- c("linear")
+           option = list(mle_solver = NULL)) {
+    supported_model <- c("linear", "logit")
 
     if (!(model %in% supported_model)) {
       stop(sprintf("The model %s is not supported.", model))
 
     }
 
-    option$mle_solver <-
-      match.arg(option$mle_solver, c("pseudo inverse", "BFGS"))
-
     coef_estimate <- NULL
 
-    if (option$mle_solver == "pseudo inverse") {
-      coef_estimate <- linear.mle.pseudo_inverse(design, outcome)
-    } else if (option$mle_solver == "BFGS") {
-      coef_estimate <- linear.mle.BFGS(design, outcome)
-    } else {
-      stop("No current plans for adding solver options other than pseudo inverse and BFGS.")
+    if (model == "linear") {
+      option$mle_solver <-
+        match.arg(option$mle_solver, c("pseudo inverse", "BFGS"))
+      if (option$mle_solver == "pseudo inverse") {
+        coef_estimate <- linear.mle.pseudo_inverse(design, outcome)
+      } else if (option$mle_solver == "BFGS") {
+        coef_estimate <- linear.mle.BFGS(design, outcome)
+      } else {
+        stop("No current plans for adding solver options other than pseudo inverse and BFGS.")
```

Comment:
```suggestion
        stop('No current plans for adding solver options other than "pseudo inverse" and "BFGS".')
```

For error messages, something like this is helpful Just so users know exactly what the required arguments are.

Though in this function match.arg will error and this line never runs, so it's not needed.